src/Project/JobRunner.ts

```diff
@@ -45,9 +45,10 @@ export class JobRunner extends EventEmitter {
     this.on(K_CLEANUP, this.onCleanup);
   }
 
-  private invoke(job: Job, path: string) {
+  private invoke(job: Job) {
     let tool: string = job.tool;
     let toolArgs: ToolArgs = job.toolArgs;
+    let workDir: string = job.workDir;
```

Comment:
How about handling if job.wordDir is empty?

Reply:
There are two cases of implementing `Job` interface.
- JobCommand
- JobBase

JobCommand class defines the `workDir` as `require('os').homedir()` function because this class usually is used by Toolchain or Device APIs to request install, uninstall or register somethings.
JobBase class defines the `workDir` as '' null string. It will be replaced by `workDir = helpers.obtainWorkspaceRoot();` value at the point when it is really executing.

As you know, we usually use `JobCommand` class and `JobBase` is a remnant of old codes.

Reply:
👍